client/quantum_serverless/utils/storage.py

```diff
@@ -0,0 +1,77 @@
+# This code is a Qiskit project.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+===========================================================
+Storage utilities (:mod:`quantum_serverless.utils.storage`)
+===========================================================
+
+.. currentmodule:: quantum_serverless.utils.storage
+
+Quantum serverless storage utilities
+====================================
+
+.. autosummary::
+    :toctree: ../stubs/
+
+    PersistentStorage
+"""
+from io import BytesIO
+import os
+from typing import Optional
+
+from abc import abstractmethod
+import s3fs
+
+
+class BaseStorage:
+    """Base class for persistent storage."""
+
+    @abstractmethod
+    def save(self, path: str, data: BytesIO):
+        """Save data."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def load(self, path: str):
+        """Load data."""
+        raise NotImplementedError
+
+
+class S3Storage(BaseStorage):
+    """Class for storing s3 objects in a non-temporary manner."""
+
+    def __init__(
+        self,
+        endpoint: str,
+        bucket: str,
+        key: Optional[str] = None,
+        secret: Optional[str] = None,
+    ):
+        """Long-term storage for serverless computation."""
+        self.endpoint = endpoint
+        self.bucket = bucket
+        self.key = key or os.getenv("AWS_ACCESS_KEY")
+        self.secret = secret or os.getenv("AWS_SECRET_ACCESS_KEY")
```

Comment:
This is an absolute suggestion/question so, thinking that S3 almost works as an standard in most of the clouds for object storage does it have sense to rename this `env_vars` to:
- `S3_ACCESS_KEY`
- `S3_SECRET_ACCESS_KEY`

With the purpose to use this class with any object storage from any cloud provider without the reference to `AWS`, what do you think @psschwei , @IceKhan13 ?

Reply:
The AWS CLI (which a lot of folks use to interact with s3 stores, including IBM COS) requires these exact envvars, so that's why I didn't pick a more generic value here

Reply:
I see, thank you for the explanation, Paul. I agree then 😄